src/Core/Repository/BaseRepositoryInterface.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Core\Repository;
+
+interface BaseRepositoryInterface
+{
+    public function save();
```

Comment:
As we discussed - it's either unit of work with data mapper or update($user) method. So, as it should be done the best way there is, let's try to bring another abstraction layer ;)